### ISSUE TITLE
Should the Dot layer have "normalize" set to True?

### ISSUE THREAD

**Issue by dlyubomirov** [keras-team/keras-io#1214]
In the "[Graph representation learning with node2vec](https://keras.io/examples/graph/node2vec_movielens/)" example, the Dot layer in the "create_model" function has the "normalize" flag set to False, see below

```
def create_model(vocabulary_size, embedding_dim):

    ...

    # Lookup embeddings for target.
    target_embeddings = embed_item(inputs["target"])
    # Lookup embeddings for context.
    context_embeddings = embed_item(inputs["context"])
    # Compute dot similarity between target and context embeddings.
    logits = layers.Dot(axes=1, normalize=False, name="dot_similarity")(
        [target_embeddings, context_embeddings]
    )

    ...
```

What is the rationale for not applying normalization? With no normalization, the output of the Dot layer would be arbitrarily large (depending on the inputs). Conversely, normalizing the inputs would squash the range of the output into the range `[-1, 1]`, which (for me) makes more sense when comparing it to the ground truth (i.e. values 0 or 1).

**Comment by rchao** [comment]
Hello, would you like to directly reach out to the author for an answer? The author is [Khalid Salama](https://www.linkedin.com/in/khalid-salama-24403144/).

**Comment by rchao** [comment]
(I assume you're referring to "Graph representation learning with node2vec" example)

**Comment by dlyubomirov** [comment]
> (I assume you're referring to "Graph representation learning with node2vec" example)

Thanks, I updated the original post.

**Comment by SuryanarayanaY** [comment]
Hi @dlyubomirov ,

In the tutorial, the model is compiled for loss function `BinaryCrossentropy` with `from_logits=True` and hence in the Dot layer it was set `normalize=False` as the loss function expects logits not normalized values. Users can change both these settings in sync i.e. if loss function have from_logits=False then Dot layer should have normalize=True and vice versa.

```
model.compile(
    optimizer=keras.optimizers.Adam(learning_rate),
    loss=keras.losses.BinaryCrossentropy(from_logits=True),
)
```

There is no specific reason for this but `from_logits=True` avoids some numerical problems. Please refer to the [comment](https://github.com/keras-team/keras-io/pull/1439#issuecomment-1639078889) here for more details which might be relevant to this issue.
